File: tags/VER0_7_0/delta3d/examples/testPython/testpython.py
from dtCore import *
from dtABC import *

from math import *
from time import *

def radians(v):
   return v * pi/180
   
class TestPythonApplication(Application):
    def Config(self):
        Application.Config(self)
        SetDataFilePathList('../../data');
        self.helo = Object('UH-1N')
        self.helo.LoadFile('UH-1N/UH-1N.ive')
        self.GetScene().AddDrawable(self.helo)
        self.transform = Transform()
        self.angle = 0.0
        
    def PreFrame(self, deltaFrameTime):
        self.transform.Set(40*cos(radians(self.angle)),
                           100 + 40*sin(radians(self.angle)), 
                           0, self.angle, 0, -45)
        self.helo.SetTransform(self.transform)
        self.angle += 45*deltaFrameTime

testPythonApp = TestPythonApplication('config.xml')

testPythonApp.Config()
testPythonApp.Run()

File: branches/BRANCH_1_0_0/delta3d/examples/testPython/testpython.cpp
#include "dtCore/dt.h"
#include "dtABC/dtabc.h"

#include <OpenThreads/Thread>
#include <boost/python/detail/wrap_python.hpp>

using namespace dtCore;
using namespace dtABC;


/**
* The Python test application.
*/
class TestPythonApp : public Application,
   public OpenThreads::Thread
{
   DECLARE_MANAGEMENT_LAYER(TestPythonApp)

public:

   /**
   * Constructor.
   */
   TestPythonApp( std::string configFile = "config.xml" )
      : Application( configFile )
   {

      Object* obj = new Object("AH-1W");

      obj->LoadFile("ah-1w.ive");

      Transform transform(0, 50, 0);

      obj->SetTransform(&transform);

      GetScene()->AddDrawable(obj);

      Py_Initialize();

      start();
   }

   /**
   * The run thread.
   */
   virtual void run()
   {
      PyRun_InteractiveLoop(stdin, "???");

      Quit();
   }
};

IMPLEMENT_MANAGEMENT_LAYER(TestPythonApp)

int main( int argc, char **argv )
{
   SetDataFilePathList("..;" + GetDeltaDataPathList());

   TestPythonApp* app = new TestPythonApp( "config.xml" );

   app->Config();
   app->Run();

   delete app;

   return 0;
}
